lib/puppetserver/ca/host.rb

```diff
@@ -1,36 +1,153 @@
 require 'openssl'
+require 'fileutils'
+require 'yaml'
 
 module Puppetserver
   module Ca
     class Host
+      # Exclude OIDs that may conflict with how Puppet creates CSRs.
+      #
+      # We only have nominal support for Microsoft extension requests, but since we
+      # ultimately respect that field when looking for extension requests in a CSR
+      # we need to prevent that field from being written to directly.
+      PRIVATE_CSR_ATTRIBUTES = [
+        'extReq',   '1.2.840.113549.1.9.14',
+        'msExtReq', '1.3.6.1.4.1.311.2.1.14',
+      ]
+
+      PRIVATE_EXTENSIONS = [
+        'subjectAltName', '2.5.29.17',
+      ]
+
+      # A mapping of Puppet extension short names to their OIDs. These appear in
+      # csr_attributes.yaml.
+      PUPPET_SHORT_NAMES =
+      {:pp_uuid =>             "1.3.6.1.4.1.34380.1.1.1",
```

Comment:
I'm not sure what we're doing about this in Puppet, but in my testing it looks like the keys parsed from the YAML is coming in as strings. Maybe we need to update these keys to be strings.

Reply:
Yep, that was a bug.  Just fixed it and added a spec.  Thank you!